app/services/jobs/return-logs/fetch-licence-return-requirements.service.js

```diff
@@ -0,0 +1,128 @@
+'use strict'
+
+/**
+ * Fetches data needed for the generating return logs
+ * @module FetchReturnLogsService
+ */
+
+const ReturnLogModel = require('../../../models/return-log.model.js')
+const ReturnRequirementModel = require('../../../models/return-requirement.model.js')
+const ReturnVersionModel = require('../../../models/return-version.model.js')
+
+const { db } = require('../../../../db/db.js')
+const { cycleEndDate, cycleStartDate, startOfWinterAndAllYearCycle } = require('../../../lib/dates.lib.js')
+
+/**
+ * Given the licence reference this service returns the return requirements to be turned into return logs.
+ *
+ * @param {string} licenceReference - if provided only do the return log for that licence
+ *
+ * @returns {Promise<Array>} the array of return log payloads to be created in the database
+ */
+async function go (licenceReference) {
+  const returnRequirements = await _fetchReturnRequirements(licenceReference)
+
+  return returnRequirements
+}
+
+async function _fetchExternalIds (licenceReference) {
+  const externalIds = await ReturnLogModel.query()
+    .select([db.raw("concat(metadata->'nald'->>'regionCode', ':', return_reference) as externalid")])
+    .where('startDate', '>=', startOfWinterAndAllYearCycle())
+    .where('licenceRef', licenceReference)
+
+  const externalIdsArray = externalIds.map((item) => {
+    return item.externalid
+  })
+
+  return externalIdsArray
+}
+
+async function _fetchReturnRequirements (licenceReference) {
+  // pass in false to ge the all year start and end date to make sure we cover the entire year
+  const _cycleEndDate = cycleEndDate(false)
+  const _cycleStartDate = cycleStartDate(false)
+  const externalIds = await _fetchExternalIds(licenceReference)
+
+  return await ReturnRequirementModel.query()
```

Comment:
We can drop the await here

```suggestion
  return ReturnRequirementModel.query()
```